go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/edgexfoundry/go-mod-core-contracts/v4
 
-go 1.23
+go 1.23.0
+
+toolchain go1.23.7
```

Comment:
do you know if this toolchain impact anything?

Reply:
The toolchain will specify the version of standard library as well as the compiler, assembler, and other tools. I suspect that dependbot fails due to missing toolchain, so just gave it a try. As I'm still investigating this issue, convert this PR to draft.

Reply:
Well, I pushed another commit https://github.com/edgexfoundry/go-mod-core-contracts/pull/997/commits/4a47f5b9bc59228376409a9280fda300c610fb5a to remove the toolchain in go.mod, and the `make test` fails at the same error as dependabot.